web/modules/custom/nfafmis/nfafmis.views.inc

```diff
@@ -37,4 +37,68 @@ function nfafmis_views_query_alter(ViewExecutable $view, QueryPluginBase $query)
       }
     }
   }
+  elseif ($view->storage->get('id') === 'cfr_lists') {
+    // Get the current user's assigned management unit.
+    $nfafmis_service_utility = \Drupal::service('nfafmis_service.utility');
+    $range = $nfafmis_service_utility->getUserManagementUnit();
+
+    if ($range) {
+      // Add a join to the management unit data table.
+      $definition = [
+        'table' => 'taxonomy_term__management_unit',
+        'type' => 'INNER',
+        'field' => 'entity_id',
+        'left_table' => 'taxonomy_term_field_data',
+        'left_field' => 'tid',
+      ];
+
+      $join = Drupal::service('plugin.manager.views.join')->createInstance('standard', $definition);
+      $query->addRelationship('taxonomy_term__management_unit', $join, 'person');
+
+      // Limit CFRs to those that are in the user's management unit.
+      $query->addWhere('conditions', 'taxonomy_term__management_unit.management_unit_target_id ', $range->id(), '=');
```

Comment:
Wouldn't be a custom filter a better tool for this job?

Reply:
Yeah, I started writing one but then decided that it's such a specific filter (only applies to the one taxonomy that has a custom management unit field) it didn't make sense to make a generic filter. Also I struggled to get it to work and wanted to get this to Rashid this week :)

Reply:
I replaced this with a custom filter NfaUserManagementUnit